torch/_decomp/decompositions.py

```diff
@@ -3053,15 +3047,21 @@ def _make_base_grid_5d(
 
 def _affine_grid_generator_4d(theta: Tensor, size: List[int], align_corners: bool):
     n, _, h, w = size
-    base_grid = _make_base_grid_4d(theta, n, h, w, align_corners=align_corners)
-    grid = base_grid.view(n, h * w, 3).bmm(theta.transpose(1, 2))
+    base_grid = _make_base_grid_4d(theta, h, w, align_corners=align_corners)
+    # base_grid shape is (h, w, 3) and theta shape is (n, 2, 3)
+    # We do manually a matrix multiplication which is faster than mm()
+    # (h * w, 3, 1) * (n, 1, 3, 2) -> (n, h * w, 2)
+    grid = (base_grid.view(-1, 3, 1) * theta.mT.unsqueeze(1)).sum(-2)
     return grid.view(n, h, w, 2)
```

Comment:
You can even do:
```suggestion
    return (base_grid.unsqueeze(-1) * theta.mT.view(n, 1, 1, 3, 2)).sum(-2)
```
No need to merge the h and w dimensions just to split them again later.